packages/matter.js/src/util/String.ts

```diff
@@ -7,3 +7,160 @@
 export function capitalize(text: string) {
     return text[0].toUpperCase() + text.slice(1);
 }
+
+/**
+ * Converts identifiers of the form "foo-bar", "foo_bar", "foo bar", "foo*bar",
+ * "fooBar" or "FOOBar" into "FooBar" or "fooBar".
+ */
+export function camelize(name: string, upperFirst = true) {
+    const pieces = new Array<string>();
+    let pieceStart = 0,
+        sawUpper = false,
+        sawLower = false,
+        i = 0;
+
+    function addPiece(to: number) {
+        if (pieceStart < to) pieces.push(name.slice(pieceStart, to));
+        sawLower = sawUpper = false;
+    }
+
+    for (; i < name.length; i++) {
+        if (name[i] >= "A" && name[i] <= "Z") {
+            if (sawLower) {
+                addPiece(i);
+                pieceStart = i;
+            }
+            sawUpper = true;
+            continue;
+        }
+
+        if (name[i] >= "a" && name[i] <= "z") {
+            if (!sawLower) {
+                if (sawUpper) {
+                    addPiece(i - 1);
+                    pieceStart = i - 1;
+                }
+            }
+            sawLower = true;
+            continue;
+        }
+
+        addPiece(i);
+
+        if ((name[i] >= "0" && name[i] <= "9") || name[i] == "$") {
+            pieces.push(name[i])
+        }
+
+        pieceStart = i + 1;
+        continue;
+    }
+    addPiece(i);
+
+    let didFirst = false;
+    return pieces.map((piece) => {
+        let firstChar = piece[0];
+        if (upperFirst || didFirst) {
+            firstChar = firstChar.toUpperCase();
+        } else {
+            firstChar = firstChar.toLowerCase();
+            didFirst = true;
+        }
+        return `${firstChar}${piece.slice(1).toLowerCase()}`;
+    }).join("");
+}
+
+/**
+ * Like JSON.stringify but targets well-formed JS and is slightly more readable.
+ */
+export function serialize(value: any) {
+    const visited = new Set();
+
+    function asValidKey(key: string) {
+        if (key.match(/[a-z_\$][a-z_\$0-9]*/i)) {
+            return key;
+        }
+        return JSON.stringify(key);
+    }
+    
+    function serializeOne(value: any): string | undefined {
+        if (value === undefined) {
+            return;
+        }
+        if (value == null) {
+            return "null";
+        }
+        if (value[serialize.SERIALIZE]) {
+            return value[serialize.SERIALIZE]();
+        }
+        if (typeof value == "function") {
+            return;
+        }
+        if (typeof value == "bigint" || value instanceof BigInt) {
```

Comment:
Please add ByteArray to be logged as hex

Reply:
OK serializes anything ArrayBuffer.isView() as hex now